packages/compass-e2e-tests/tests/connection-form.test.ts

```diff
@@ -486,4 +487,84 @@ describe('Connection form', function () {
       readPreference: 'defaultReadPreference',
     });
   });
+
+  it('can save a connection as a favorite and manage it', async function () {
+    const favoriteName = 'My Favorite';
+    const newFavoriteName = 'My Favorite (edited)';
+
+    // save
+    await browser.clickVisible(Selectors.ConnectionFormEditFavouriteButton);
+    await browser.$(Selectors.FavoriteModal).waitForDisplayed();
+    await browser.$(Selectors.FavoriteNameInput).setValue(favoriteName);
+    await browser.clickVisible(
+      `${Selectors.FavoriteColorSelector} [data-testid="color-pick-color1"]`
+    );
+    await browser.$(Selectors.FavoriteSaveButton).waitForEnabled();
+    await browser.clickVisible(Selectors.FavoriteSaveButton);
+    await browser.$(Selectors.FavoriteModal).waitForExist({ reverse: true });
+
+    // HACK: The favorite doesn't always appear in the sidebar. Must be some
```

Comment:
I suspect some of the other flakes could be a similar thing and we can probably work the sidebar database/collection flakes the same way.

But this is still a bug in the app.

Reply:
Can it be the fact that save is not currently awaited?

Reply:
I don't know but it _does_ show up when you reload the app, so it did save. It just never shows up until you reload.

My gut suspects that if you save a connection at just the right time while connections are loaded then the connections get returned without that connection.

Reply:
I'm awaiting it now. Let's see if that helps.